File: MATLAB_Prototype/Tests/dt_IIDDistribution.m
%% dt_IIDDistribution
% doctest for IIDDistribution
%
%  Examples
%
%  Example 1:
%
% >> iid = IIDDistribution
% 
%   iid =***
%     
% 
%          distribData: []
%                state: []
%             nStreams: 1
%               domain: [2x2 double]
%           domainType: 'box'
%            dimension: 2
%     trueDistribution: 'uniform'
%
%  
% >> iid2 = iid.initStreams(1)
% 
%   iid2 =*** 
% 
%   IIDDistribution with properties:
% 
%          distribData: [1x1 struct]
%                state: []
%             nStreams: 1
%               domain: [2x2 double]
%           domainType: 'box'
%            dimension: 2
%     trueDistribution: 'uniform'
%
%
% >> iid2.distribData.stream{1,1}
% 
%     mrg32k3a random stream
%                  Seed: 0
%       NormalTransform: Ziggurat
%
%   
% >> rng(100);
% >> [x, w, a] = iid2.genDistrib(1, 2, 3, [1 2])
% 
%     x =
% 
%         0.7270    0.9387
%         0.4522    0.2360
% 
%     w =
% 
%          1
% 
%     a =
% 
%         0.3333
%

File: Python_Prototype/dt_IIDDistribution.py
>>> from IIDDistribution import IIDDistribution;iid = IIDDistribution();print(iid.__dict__)
{'domain': array([[0, 0],
        [1, 1]]), 'domainType': 'box', 'dimension': 2, 'trueDistribution': 'uniform'}

>>> iid2 = iid.initStreams(1);print(iid2.__dict__)
{'domain': array([[0, 0],
        [1, 1]]), 'domainType': 'box', 'dimension': 2, 'trueDistribution': 'uniform', 'distribDataStream': [...]}


>>> x,w,a = iid2.genDistrib(1, 2, 3, (1,2));print(iid2.__dict__)
{'domain': array([[0, 0],
    [1, 1]]), 'domainType': 'box', 'dimension': 2, 'trueDistribution': 'uniform', 'distribDataStream': [<mtrand.RandomState object at 0x000002737F55DAF8>]}
>>> print(x,w,a)
1 0.3333333333333333
